Packs/ZeroNetworks/Integrations/ZeroNetworksEventCollector/ZeroNetworksEventCollector.py

```diff
@@ -0,0 +1,204 @@
+MAX_LIMIT = 10000
+LOG_TYPE = ['audit', 'network_activities']
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    API_VERSION = "/api/v1"
+
+    def __init__(self, server_url: str, proxy: bool, verify: bool, headers: dict):
+        super().__init__(
+            base_url=urljoin(server_url, self.API_VERSION),
+            verify=verify,
+            proxy=proxy,
+            headers=headers
+        )
+
+    def search_events(self, limit, cursor) -> dict[str, Any]:
+        """
+        Get a list of events.
+        Args:
+            start_date (str, optional): Fetch events that are newer than given time.
+                Defaults to None.
+            limit (int, optional): Maximum number of events to return.
+                Defaults to None.
+            offset (int, optional): Item number to start looking from.
+                Defaults to None.
+        Returns:
+            Dict[str, Any]: A list of events.
+        """
+        params = remove_empty_elements({"_limit": limit, "order": "asc", "_cursor": cursor})
+
+        return self._http_request(
+            method="GET",
+            url_suffix="/audit",
+            params=params,
+        )
+
+
+''' HELPER FUNCTIONS '''
+
+
+def process_events(events: list, previous_ids: set, last_event_time: int, max_results: int, num_results: int):
+    new_events = []
+    for event in events:
+        event_id = event.get('id')
+        if num_results == max_results:
+            break
+        
+        if event_id not in previous_ids:
+            event['_TIME'] = timestamp_to_datestring(event.get('timestamp'), is_utc=True)
+            event['source_log_type'] = 'audit'
+            new_events.append(event)
+            event_timestamp = event.get("timestamp")
+            num_results += 1
+            if event_timestamp > last_event_time:
+                demisto.debug('updating the last run')
+                previous_ids = {event_id}
+                last_event_time = event_timestamp
+
+            # Adding the event ID when the event time is equal to the last received event
+            elif event_timestamp == last_event_time:
+                demisto.debug('adding id to the "new_previous_ids"')
+                previous_ids.add(event_id)
+
+    return new_events, previous_ids, last_event_time, num_results
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def fetch_events(client: Client, params: dict, last_run: dict):
+    """
+       Fetches audit logs from ZeroNetworks API.
+    """
+    start_timestamp = last_run.get("last_fetch")
+    if not start_timestamp:
+        start_date = dateparser.parse(FIRST_FETCH).strftime(ISO_8601_FORMAT)   # type: ignore[union-attr]
+        start_timestamp = date_to_timestamp(start_date, ISO_8601_FORMAT)
+
+    cursor = start_timestamp
+    last_event_time = start_timestamp
+    collected_events: list = []
+    max_results: int = arg_to_number(params.get('max_results')) or MAX_LIMIT
+    previous_ids = last_run.get('previous_ids', ())
+    num_results = 0
```

Comment:
Please remove this num_results and use instead len(collected_events) for the value inserted to process_events(...) method.